CI/integration_tests/transformations/test_molecular_mapping_results.py

```diff
@@ -0,0 +1,246 @@
+                directory.
+
+        Returns
+        -------
+        Tests that the molecule groups detected are done so correctly and that the
+        constructed trajectory is also correct.
+        """
+        mdsuite_project.experiments["ligand_water"].species["OW"].mass = [15.999]
+        mdsuite_project.experiments["ligand_water"].species["HW1"].mass = [1.00784]
+        mdsuite_project.experiments["ligand_water"].species["HW2"].mass = [1.00784]
+        reference_molecules = {
+            "water": {
+                "n_particles": 14,
+                "mass": 18.014680000000002,
+                "groups": {
+                    "0": {"HW1": [0], "OW": [0], "HW2": [0]},
+                    "1": {"HW1": [1], "OW": [1], "HW2": [1]},
+                    "2": {"HW1": [2], "OW": [2], "HW2": [2]},
+                    "3": {"HW1": [3], "OW": [3], "HW2": [3]},
+                    "4": {"HW1": [4], "OW": [4], "HW2": [4]},
+                    "5": {"HW1": [5], "OW": [5], "HW2": [5]},
+                    "6": {"HW1": [6], "OW": [6], "HW2": [6]},
+                    "7": {"HW1": [7], "OW": [7], "HW2": [7]},
+                    "8": {"HW1": [8], "OW": [8], "HW2": [8]},
+                    "9": {"HW1": [9], "OW": [9], "HW2": [9]},
+                    "10": {"HW1": [10], "OW": [10], "HW2": [10]},
+                    "11": {"HW1": [11], "OW": [11], "HW2": [11]},
+                    "12": {"HW1": [12], "OW": [12], "HW2": [12]},
+                    "13": {"HW1": [13], "OW": [13], "HW2": [13]},
+                },
+            }
+        }
+        water_molecule = mdsuite.Molecule(
+            name="water",
+            species_dict={"OW": 1, "HW1": 1, "HW2": 1},
+            amount=14,
+            cutoff=1.7,
+            mol_pbc=True,
+        )
+        mdsuite_project.experiments["ligand_water"].run.MolecularMap(
+            molecules=[water_molecule]
+        )
+        molecules = mdsuite_project.experiments["ligand_water"].molecules
+        assert molecules == reference_molecules
+
+        assert "water" not in mdsuite_project.experiments["ligand_water"].species
+
+    def test_ionic_liquid(self, mdsuite_project):
+        """
+        Test molecule mapping on a more complex ionic liquid.
+
+        This test will ensure that one can pass multiple molecules to the mapper as
+        well as check the effect of parsing a specific reference configuration.
+        """
+        bmim_molecule = mdsuite.Molecule(
+            name="bmim",
+            species_dict={"C": 8, "N": 2, "H": 15},
+            amount=50,
+            cutoff=1.9,
+            reference_configuration=100,
+        )
+        bf_molecule = mdsuite.Molecule(
+            name="bf4",
+            smiles="[B-](F)(F)(F)F",
+            amount=50,
+            cutoff=2.4,
+            reference_configuration=100,
+        )
+        mdsuite_project.experiments["bmim_bf4"].run.MolecularMap(
+            molecules=[bmim_molecule, bf_molecule]
+        )
```

Comment:
We can, yes I will add one. The initial reason behind not doing that is because the molecules in this sim are not set so assuredly as those in the water ones. However, the computed RDF is mapping result is correct so these molecules are mapped correctly. They also pass al isomorphism checks implemented.

Reply:
Resolved in 836ea79